src/runtime/converter.cs

```diff
@@ -701,16 +701,12 @@ private static bool ToPrimitive(IntPtr value, Type obType, out object result, bo
                         op = value;
                         if (Runtime.PyObject_TYPE(value) != Runtime.PyLongType)
                         {
-                            op = Runtime.PyNumber_Long(value);
-                            if (op == IntPtr.Zero)
-                            {
-                                goto convert_error;
-                            }
+                            goto type_error;
```

Comment:
I think this can be skipped alltogether. https://docs.python.org/3.8/c-api/long.html#c.PyLong_AsUnsignedLongLong already checks the type.